src/com/epam/javatrainings/annotations/validation/EmailAnnotationProcessor.java

```diff
@@ -0,0 +1,25 @@
+package com.epam.javatrainings.annotations.validation;
+
+import java.lang.annotation.Annotation;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class EmailAnnotationProcessor extends AnnotationProcessor {
+    @Override
+    String validate(Annotation annotation, Object value) {
+        if (!(value instanceof String)) {
+            throw new IllegalStateException();
+        }
+
+        final String emailRegex = "^[a-zA-Z0-9_!#$%&'*+/=?`{|}~^.-]+@[a-zA-Z0-9.-]+$";
```

Comment:
this can be a constant field in the class.

Reply:
Thanks for the feedback, all the changes are done and pushed.